src/generic_core/uproxy_core.ts

```diff
@@ -638,138 +671,93 @@ export class uProxyCore implements uproxy_core_api.CoreApi {
+              CLOUD_DEPLOY_PROGRESS + (progress * ((100 - CLOUD_DEPLOY_PROGRESS) / 100)));
           });
 
-          // TODO: The provisioning module should return the username!
           return installer.install(host, port, 'root', serverInfo.ssh.private);
         }).then((cloudNetworkData: any) => {
-          // TODO: make cloudNetworkData an Invite type.  This requires the cloud
-          // social provider to export the Invite interface, and also to cleanup
-          // reference paths so that uProxy doesn't need to include modules like
-          // ssh2.
-          destroyModules();
-
-          // Login to the Cloud network if the user isn't already so that
-          // we can add the new cloud server to the roster.
-          return this.loginIfNeeded_('Cloud').then((cloudNetwork) => {
-            // Set flag so Cloud social provider knows this invite is for the admin
-            // user, who just created the server.
-            cloudNetworkData['isAdmin'] = true;
-
-            // Synthesize an invite token based on cloudNetworkData.
-            // CONSIDER: if we had a social.acceptInvitation that only took
-            // networkData we wouldn't need to fake the other fields.
-            return cloudNetwork.acceptInvitation({
-              v: 2,
-              networkName: 'Cloud',
-              userName: cloudNetworkData['host'],
-              networkData: JSON.stringify(cloudNetworkData)
-            });
-          });
-        }, (installError: any) => {
-          // When we cancel the cloud install we close the connection
-          // to the installer by destroying the modules
-          if (installError === 'closed') {
-            return Promise.reject(new Error('canceled'));
-          }
-          
-          // Tell user if the server already exists
-          if (installError.errcode === "VM_AE") {
-            destroyModules();
-            return Promise.reject(new Error('server already exists'));
-          }
-
-          // Destroy the server we just created so that the user isn't billed.
-          log.error('install failed, cleaning up');
-          return provisioner.stop(DROPLET_NAME).then((unused: Object) => {
-            log.info('destroyed server on digitalocean');
-            destroyModules();
-            return Promise.reject(installError);
-          }, (destroyError: Error) => {
-            // This is bad: the user will be billed for a broken server.
-            // TODO: direct the user at the digitalocean console
-            log.error('failed to destroy new server after install failure: %1',
-              destroyError.message);
-            destroyModules();
-            return Promise.reject(installError);
+          // Set flag so Cloud social provider knows this invite is for the admin
+          // user, who just created the server.
+          cloudNetworkData['isAdmin'] = true;
+
+          // TODO: what if this fails?
+          return cloudNetwork.acceptInvitation({
```

Comment:
> // TODO: what if this fails?

(I assume that either we don't expect failure to happen often enough, or that the failure is recoverable enough, that we shouldn't block merging on handling it here. If that's right, should the comment say something a little less scary, like `TODO: Handle failure (unlikely)`, or is this actually scary?